subcell_pipeline/simulation/cytosim/run_cytosim_compression_batch_simulations.py

```diff
@@ -0,0 +1,184 @@
+Notebook contains steps for running Cytosim simulations in which a single actin
+fiber is compressed at different compression velocities.
+
+This notebook uses [Cytosim](https://github.com/simularium/Cytosim) templates
+and scripts. Clone a copy and set the environment variable
+`CYTOSIM=/path/to/Cytosim/`.
+
+This notebook provides an example of running a simulation series in which there
+are multiple conditions, each of which need to be run for multiple replicates.
+For an example of running a simulation series for a single condition for
+multiple replicates, see `run_cytosim_single_fiber_batch_simulations.py`.
+
+- [Define simulation conditions](#define-simulation-conditions)
+- [Generate configs from template](#generate-configs-from-template)
+- [Define simulation settings](#define-simulation-settings)
+- [Create and run jobs](#create-and-run-jobs)
+"""  # noqa: D400, D415
+
+# %%
+import getpass
+import os
+import sys
+from datetime import datetime
+from pathlib import Path
+
+from dotenv import load_dotenv
+
+from subcell_pipeline.simulation.batch_simulations import (
+    check_and_save_job_logs,
+    generate_configs_from_template,
+    register_and_run_simulations,
+)
+
+# %%
+load_dotenv()
+
+# %% [markdown]
+"""
+Template generator uses the Preconfig class from Cytosim. Append the path to
+`preconfig.py` from a copy of the `Cytosim` repository, or download a copy into
+this location.
+"""
+
+# %%
+sys.path.append(str(Path(os.getenv("CYTOSIM")) / "python" / "run"))
+from preconfig import Preconfig  # noqa: E402
+
+# %% [markdown]
+"""
+## Define simulation conditions
+
+Defines the `COMPRESSION_VELOCITY` simulation series, which compresses a single
+500 nm actin fiber at four different velocities (4.7, 15, 47, and 150 $\mu m$/s)
+with five replicates each (random seeds 1, 2, 3, 4, and 5).
+"""
+
+# %%
+# Name of the simulation series
+series_name: str = "COMPRESSION_VELOCITY"
+
+# S3 bucket for input and output files
+bucket: str = "s3://cytosim-working-bucket"
+
+# Random seeds for simulations
+random_seeds: list[int] = [1, 2, 3, 4, 5]
+
+# Path to the config template file
+path_to_template: str = str(
+    Path(os.getenv("CYTOSIM")) / "templates" / "vary_compress_rate.cym.tpl"
+)
+
+# Current timestamp used to organize input and outfile files
+timestamp: str = datetime.now().strftime("%Y-%m-%d")
+
+# File keys for each velocity
+velocity_keys: dict[str, str] = {
+    "4.73413649": "0047",
+    "15": "0150",
+    "47.4341649": "0470",
+    "150": "1500",
+}
+
+# %% [markdown]
+"""
+## Generate configs from template
+
+Use the `Preconfig` from Cytosim to convert the `.cym.tpl` template file into
+local `.cym` config files. For each file, extract the compression velocity to
+use as the simulation condition key. Save all config files to S3 bucket.
+"""
+
+# %%
+preconfig = Preconfig()
+config_files = preconfig.parse(path_to_template, {})
+
+# %%
+pattern = r"compression_velocity:([\s0-9\.]+)"
+group_keys = generate_configs_from_template(
```

Comment:
Getting an error: `An error occurred (AccessDenied) when calling the PutObject operation: Access Denied` at this step without a prompt for AWS credentials. Is there a way I could set those up?

Reply:
Yeah, do you want to try troubleshooting this at our infra meeting tomorrow?